pkg/repository.go

```diff
@@ -0,0 +1,76 @@
+package pkg
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	log "github.com/sirupsen/logrus"
+)
+
+//struct which represent mainly interaction with redis
+type RedisRepository struct {
+	Client *redis.Client
+	Ctx  context.Context
+}
+
+
+//should update existing history with order or create a new one 
+func (r *RedisRepository) SaveOrder(userId string, order Order) error {
+	val, err := r.Client.Get(r.Ctx, (userId)).Result()
+	if err != nil {
```

Comment:
if we failed, don't try to do some special logic, fixing, etc. Just return error.
Actually, if we failed on `Get` then it probably will fail and on `Set` inside `saveHistory`

Reply:
I did a mistake here, instead of checking if user is nil, I checked the error.

The idea of this function is if user is nil just create an new entry with user id the order